ios/RCTOneSignal/RCTOneSignalEventEmitter.m

```diff
@@ -173,6 +154,54 @@ + (void)sendEventWithName:(NSString *)name withBody:(NSDictionary *)body {
     }
 }
 
+RCT_REMAP_METHOD(getDeviceState,
+                getDeviceStateResolver:(RCTPromiseResolveBlock)resolve
+                rejecter:(RCTPromiseRejectBlock)reject) {
+    OSDeviceState *deviceState = [OneSignal getDeviceState];
+
+    if (deviceState) {
+        resolve([deviceState jsonRepresentation]);
+    } else {
+        NSError * error = nil;
+        NSString * message = @"Could not get OneSignal device state";
+        reject(@"no_value", message, error);
+    }
+}
+
+RCT_EXPORT_METHOD(setNotificationOpenedHandler) {
+    [OneSignal setNotificationOpenedHandler:^(OSNotificationOpenedResult *result) {
+        [RCTOneSignalEventEmitter sendEventWithName:@"OneSignal-remoteNotificationOpened" withBody:[result jsonRepresentation]];
+    }];
+}
+
+RCT_EXPORT_METHOD(setNotificationWillShowInForegroundHandler) {
+    [OneSignal setNotificationWillShowInForegroundHandler:^(OSNotification *notif, OSNotificationDisplayResponse completion) {
+        self->_receivedNotificationCache[notif.notificationId] = notif;
+        self->_notificationCompletionCache[notif.notificationId] = completion;
```

Comment:
Did Xcode require you use self-> here? I don't think we need the self arrow you should just be able to use the instance variable. Other than that it looks good!

Reply:
It says: `Block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior` @emawby